### PR TITLE
Fix duplicated Astro and Vite injected styles

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Fix https://github.com/withastro/astro/issues/8637

The issue was that `<style data-astro-dev-id="..."` and `<style data-vite-dev-id="..."` have different IDs that they don't match and deduplicate correctly, causing duplicated styles. The main issue is that we're using `importedModule.url` instead of `importedModule.id`.

But I made a complete cleanup of CSS HMR, relying on Vite to handle it entirely. A lot of HMR handling was added by https://github.com/withastro/astro/pull/4157 before. It's possible to take this new approach today because Vite natively treats `data-vite-dev-id` as FOUC-prevention injected styles. 

## Testing

Existing CSS and CSS HMR tests should still work. Updated them where necessary for the slightly different markup. I also tested the https://github.com/withastro/astro/issues/8637 repro manually.

## Docs

n/a. internal change and bug fix.

### CODE REVIEW

File: packages/astro/e2e/css-sourcemaps.test.js
Comment (bluwy) [inline packages/astro/e2e/css-sourcemaps.test.js:1]: This test file seems to duplicate the same test in `css.test.js`

File: packages/astro/e2e/view-transitions.test.js [hunk 1]
```diff
@@ -631,7 +631,7 @@ test.describe('View Transitions', () => {
 	});
 
 	test('client:only styles are retained on transition', async ({ page, astro }) => {
-		const totalExpectedStyles = 8;
+		const totalExpectedStyles = 7;
```
Comment (bluwy) [inline packages/astro/e2e/view-transitions.test.js:634]: There's no more `<style data-astro-dev-id` as it's merged into `<style data-vite-dev-id`, so this number is reduced by one.

File: packages/astro/src/runtime/client/hmr.ts [hunk 2]
```diff
@@ -1,46 +1,7 @@
...
-				if (isViteInjectedStyle(node)) {
-					injectedStyles.get(node.getAttribute('data-vite-dev-id')!)?.remove();
-				}
-			}
-		}
-	});
-	mo.observe(document.documentElement, { subtree: true, childList: true });
-
-	// Vue `link` styles need to be manually refreshed in Firefox
-	import.meta.hot.on('vite:beforeUpdate', async (payload) => {
```
Comment (bluwy) [inline packages/astro/src/runtime/client/hmr.ts:20]: I manually tested this and it seems to not apply anymore, so I removed it. The rest of this file relates to our CSS handling, which is no longer needed as Vite handles all now, so pretty much the entire file is empty and not need to be loaded anymore.

**Review by matthewp — APPROVED** [review]
Amazing, thanks!

**Pull request merged**